### PR TITLE
Edge to edge in main branch

### PR DESCRIPTION
**Pull request by android-dev-lxl**
PR to test status bar transparency. 
Approved by SME.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -22,8 +22,8 @@ buildscript {
 }
 
 plugins {
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
+    id("com.android.application") version "8.1.0" apply false
+    id("com.android.library") version "8.1.0" apply false
     id("org.jetbrains.kotlin.android") version "1.8.21" apply false
```
Comment (johnshea) [inline build.gradle.kts:27]: Upgrade to latest. Latest is currently 1.9.0.

**Review by johnshea — APPROVED** [review]
LGTM - upgrade versions to latest

- Upgrade Kotlin compiler version. Latest version is currently 1.5.1

**Review by johnshea — CHANGES_REQUESTED** [review]
FAB button is overlapping the nav bar in landscape.

Screenshot 2023-09-11 at 12 27 01 PM

Screenshot 2023-09-11 at 12 26 53 PM

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt [hunk 2]
```diff
@@ -103,31 +102,35 @@ fun HomeScreen(
         HomeBody(
             itemList = homeUiState.itemList,
             onItemClick = navigateToItemUpdate,
-            modifier = modifier
-                .padding(innerPadding)
-                .fillMaxSize()
+            contentPadding = innerPadding,
```
Comment (johnshea) [inline app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:105]: nit: As you are adding contentPadding to the function signature, put contentPadding after the modifier to be consistent with the new function signature.

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt [hunk 3]
```diff
@@ -103,31 +102,35 @@ fun HomeScreen(
...
-        modifier = modifier
+        modifier = modifier,
     ) {
         if (itemList.isEmpty()) {
             Text(
                 text = stringResource(R.string.no_item_description),
                 textAlign = TextAlign.Center,
-                style = MaterialTheme.typography.titleLarge
+                style = MaterialTheme.typography.titleLarge,
+                modifier = Modifier.padding(contentPadding),
```
Comment (johnshea) [inline app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:127]: nit: modifier should be the 2nd argument to match Text composable function signature

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt [hunk 4]
```diff
@@ -136,9 +139,12 @@ private fun HomeBody(
...
 @Composable
 private fun InventoryList(
-    itemList: List<Item>, onItemClick: (Item) -> Unit, modifier: Modifier = Modifier
+    itemList: List<Item>,
+    onItemClick: (Item) -> Unit,
+    contentPadding: PaddingValues,
+    modifier: Modifier = Modifier
 ) {
-    LazyColumn(modifier = modifier) {
+    LazyColumn(contentPadding = contentPadding, modifier = modifier) {
```
Comment (johnshea) [inline app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:147]: nit: contentPadding should be after the modifier to match the function signature for LazyColumn

Also consider adding line breaks.

**Review by johnshea — APPROVED** [review]
lgtm with some suggestions.

Note - FAB is still partially under 3 button navigation bar but that is a framework bug so approving.

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt (hunk 2 above)
Comment (android-dev-lxl) [reply app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:105]: The reason for this is Modifier should be the first optional parameter, but for function calls we usually send Modifier as the last parameter. I checked some samples and they are inconsistent. 

https://github.com/android/compose-samples/blob/2410841f0cd2384789be0ad48528cabfbf097db2/Reply/app/src/main/java/com/example/reply/ui/components/ReplyEmailThreadItem.kt#L95

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt (hunk 4 above)
Comment (android-dev-lxl) [reply app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:147]: Skipping line breaks.

File: app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt (hunk 3 above)
Comment (android-dev-lxl) [reply app/src/main/java/com/example/inventory/ui/home/HomeScreen.kt:127]: Let's leave modifier at the end in function calls

**Comment by android-dev-lxl** [comment]
Added new E2E API

File: app/src/main/java/com/example/inventory/ui/item/ItemDetailsScreen.kt [hunk 5]
```diff
@@ -84,19 +90,25 @@ fun ItemDetailsScreen(
...
             )
-        }, floatingActionButton = {
+        },
+        floatingActionButton = {
             FloatingActionButton(
                 onClick = { navigateToEditItem(uiState.value.itemDetails.id) },
                 shape = MaterialTheme.shapes.medium,
-                modifier = Modifier.padding(dimensionResource(id = R.dimen.padding_large))
+                modifier = Modifier
+                        .padding(
```
Comment (johnshea) [inline app/src/main/java/com/example/inventory/ui/item/ItemDetailsScreen.kt:99]: nit: This code is indented one level too many.

**Review by johnshea — CHANGES_REQUESTED** [review]
- Check other branches to see if same issues exist in this branch.
- Status bar color in this branch is different than then other two branches. Please check and fix as needed.

**Review by johnshea — CHANGES_REQUESTED** [review]
Multiple issues still exist that were noted on other PRs and not addressed.

**Comment by android-dev-lxl** [comment]
> Multiple issues still exist that were noted on other PRs and not addressed.
As discussed please review E2E specific issues.

**Review by johnshea — APPROVED** [review]

**Pull request merged**